RFS/src/testFixtures/java/com/rfs/framework/ElasticsearchContainer.java

```diff
@@ -2,39 +2,77 @@
 
 import java.io.File;
 import java.time.Duration;
+import java.util.Map;
 
+import lombok.EqualsAndHashCode;
 import lombok.extern.slf4j.Slf4j;
 import org.testcontainers.containers.GenericContainer;
 import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableMap;
 import org.testcontainers.utility.DockerImageName;
 
 /**
  * Containerized version of Elasticsearch cluster
  */
 @Slf4j
-public class ElasticsearchContainer implements AutoCloseable {
+public class ElasticsearchContainer extends GenericContainer<ElasticsearchContainer> {
```

Comment:
Want to rename this to something like `SearchClusterContainer` and then delete the OpenSearch variant?